### ISSUE TITLE
Paywall: error 500 when I add a website

### ISSUE THREAD

**Issue by Darathor** [wallabag/wallabag#3477]
Hi,

### Issue details

After an update to Wallabag 2.3 (without error), I tried to add access to websites with paywall (I tried NextInpact and ArretSurImages) and I obtain an error 500: 

> 500: Internal Server Error
> Encoded data is shorter than expected.

In the logs, I found these lines:
```
request.INFO: Matched route "site_credentials_new". {"route":"site_credentials_new","route_parameters":{"_controller":"Wallabag\\CoreBundle\\Controller\\SiteCredentialController::newAction","_route":"site_credentials_new"},"request_uri":"http://wallabag.darathor.net/site-credentials/new","method":"POST"} []
security.DEBUG: Read existing security token from the session. {"key":"_security_secured_area"} []
security.DEBUG: User was reloaded from a user provider. {"username":"[...]","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
app.DEBUG: Crypto: crypting value: ******* [] []
request.CRITICAL: Uncaught PHP Exception Defuse\Crypto\Exception\BadFormatException: "Encoded data is shorter than expected." at [...]/wallabag/vendor/defuse/php-encryption/src/Encoding.php line 226 {"exception":"[object] (Defuse\\Crypto\\Exception\\BadFormatException(code: 0): Encoded data is shorter than expected. at [...]/wallabag/vendor/defuse/php-encryption/src/Encoding.php:226)"} []
security.DEBUG: Stored the security token in the session. {"key":"_security_secured_area"} []
```

I tried to clear caches but it didn't change anything.

### Environment

* wallabag version (or git revision) that exhibits the issue: **2.3**
* How did you install wallabag? Via `git clone` or by downloading the package? **downloading the package**
* Last wallabag version that did not exhibit the issue (if applicable): **N/A**
* php version: **7.0**
* OS: **Ubuntu 16.04**
* type of hosting (shared or dedicated): **dedicated**
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): **MySQL**

### Steps to reproduce/test case

1. click on "Accès aux sites" in the menu
2. click on the button "Créer un accès à un site"
3. fill the fields (with correct or incorrect information, the result is the same)
4. click on the button "Enregistrer"

(I don't know the english texts)

**PS :** I sometimes had file permission issues (my server installation is not perfect, I'm developper, not sysadmin) but I don't think that's it, according to the logs.

**Comment by j0k3r** [comment]
Interesting issue.
Did you got that problem on each website you added or only one? Can you try to add a random website just to be sure it's not a random issue?

**Comment by biva** [comment]
Same issue here, fresh install of 2.3 on Debian stretch

> [2017-12-12 15:27:19] request.INFO: Matched route "site_credentials_new". {"route":"site_credentials_new","route_parameters":{"_controller":"Wallabag\\CoreBundle\\Controller\\SiteCredentialController$
[2017-12-12 15:27:19] security.DEBUG: Read existing security token from the session. {"key":"_security_secured_area"} []
[2017-12-12 15:27:19] security.DEBUG: User was reloaded from a user provider. {"username":"sisim","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
[2017-12-12 15:27:19] app.DEBUG: Crypto: crypting value: n*****m [] []
[2017-12-12 15:27:19] request.CRITICAL: Uncaught PHP Exception Defuse\Crypto\Exception\BadFormatException: "Encoded data is shorter than expected." at /var/www/html/wallabag/vendor/defuse/php-encrypt$
[2017-12-12 15:27:19] security.DEBUG: Stored the security token in the session. {"key":"_security_secured_area"} []

**Comment by biva** [comment]
Tested with mediapart, lemonde, and random websites

**Comment by j0k3r** [comment]
@biva great, thanks for the feedback

**Comment by biva** [comment]
As @Darathor mentions it, concerning access rights, I changed the owner to www-data only for `web`, `var` and `vendor` with this command: `sudo chown -R www-data:www-data web var vendor`
Other folders' owner is the standard user (not root) of Debian. Should I change the owner for other folders?

**Comment by j0k3r** [comment]
Yeah, add `data` folder too.
Then could you share what's inside `data/site-credentials-secret-key.txt` (if the file exist)?

**Comment by biva** [comment]
Changing the owner of `data` is a good start: I'm able to save credentials. But if I try to edit it, user and password fields are empty.

In addition, when I try to save an article using the + button on the web UI, I get this message : `
500: Internal Server Error Parse error: syntax error, unexpected ';', expecting ']'` for this article: https://www.mediapart.fr/journal/france/121217/financement-du-terrorisme-les-15-millions-de-dollars-suspects-de-lafarge

`data/site-credentials-secret-key.txt` exists and there's one line inside, with a lot of letters and numbers.

Logs:
`[2017-12-12 16:12:50] request.INFO: Matched route "new_entry". {"route":"new_entry","route_parameters":{"_controller":"Wallabag\\CoreBundle\\Controller\\EntryController::addEntryFormAction","_route":"new_en$
[2017-12-12 16:12:50] security.DEBUG: Read existing security token from the session. {"key":"_security_secured_area"} []
[2017-12-12 16:12:50] security.DEBUG: User was reloaded from a user provider. {"username":"sisim","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
[2017-12-12 16:12:50] app.DEBUG: Restricted access config enabled? {"enabled":1} []
[2017-12-12 16:12:50] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Parse error: syntax error, unexpected ';', expecting ']'" at /var/www/html/wallabag/ven$
[2017-12-12 16:12:50] security.DEBUG: Stored the security token in the session. {"key":"_security_secured_area"} []
`

**Comment by j0k3r** [comment]
If you are able to save a new credential is means the main problem is fixed.
As far as I remember login & password are empty on editing because values are crypted in database. Should we at least decrypt the username and display it? (poke @wallabag/core)

You log seems truncated in the most interesting part :disappointed: 
```
ed ';', expecting ']'" at /var/www/html/wallabag/ven$
```

**Comment by j0k3r** [comment]
The data folder should already be writable by the www user as stated in the doc: https://doc.wallabag.org/en/admin/installation/rightaccess.html

**Comment by biva** [comment]
Sorry, I changed the permissions for every needed folders: `sudo chown -R www-data:www-data bin app/config vendor data web var`

I think you're right, we should at least decrypt the username and display it.

Sorry for the truncated logs:
`[2017-12-12 16:12:50] security.DEBUG: Read existing security token from the session. {"key":"_security_secured_area"} []
[2017-12-12 16:12:50] security.DEBUG: User was reloaded from a user provider. {"username":"sisim","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
[2017-12-12 16:12:50] app.DEBUG: Restricted access config enabled? {"enabled":1} []
[2017-12-12 16:12:50] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Parse error: syntax error, unexpected ';', expecting ']'" at /var/www/html/wallabag/vendor/j0k3r/graby/src/Extractor/HttpClient.php line 41 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Parse error: syntax error, unexpected ';', expecting ']' at /var/www/html/wallabag/vendor/j0k3r/graby/src/Extractor/HttpClient.php:41)"} []
[2017-12-12 16:12:50] security.DEBUG: Stored the security token in the session. {"key":"_security_secured_area"} []
[2017-12-12 16:57:50] request.INFO: Matched route "new_entry". {"route":"new_entry","route_parameters":{"_controller":"Wallabag\\CoreBundle\\Controller\\EntryController::addEntryFormAction","_route":"new_entry"},"request_uri":"https://bag.kumbu.eu/new-entry","method":"POST"} []
`

**Comment by nicosomb** [comment]
`Parse error: syntax error, unexpected ';', expecting ']'" at`

@biva did you change anything in ``/var/www/html/wallabag/vendor/j0k3r/graby/src/Extractor/HttpClient.php`?

**Comment by j0k3r** [comment]
I'm closing the issue for now since the problem was related to permission on folder.
We are going to write that down on the release note of 2.3.1 so user won't have the problem.
Feel free to open an other issue about your bug @biva

**Comment by biva** [comment]
@nicosomb my bad: I modified HttpClient.php and I did a mistake with some `;` that shouldn't be there. Everything is working fine now. Sorry for disturbing you guys, this issue was due to mistakes on my side.

And congrats for your work!

**Comment by Darathor** [comment]
Adding writing permission on `data` directory effectively solved the problem. Thanks

One point could be enchanced: throw an exception more in adequation with the real problem (wrinting in a not writable directory can be detected).

**Comment by MoSattler** [comment]
FYI, this is still an issue. Used the official unraid image, and the problem still remains there.

**Comment by nigiord** [comment]
I had a similar issue (status 500 error for paywal websites). Apparently it was also a permission issues, for some reasons `data/site-credentials-secret-key.txt` had owner and groups set to `wallabag2`. I solved it by using:

```bash
# In the wallabag2 install dir, for yunohost it's /var/www/wallabag2/
chgrp www-data data/site-credentials-secret-key.txt
chmod g+rw data/site-credentials-secret-key.txt
# Update credentials in wallabag interface, then
chmod g-rw data/site-credentials-secret-key.txt
```
Now it works fine. I'm not sure if I should have kept the `rw` permission for www-data group though.

**Comment by oldDuB98** [comment]
Hello everyone,
Using the version : 2.6.14 deployed on docker with Portainer, I still get the same error.
With my few knowledge on Linux, I tried to fix following your suggestion about permissions (chmod 664)  but still the same.
NB : the file "site-credentials-secret-key.txt" wasn't created during program installation nor after having activated the paywall parameter (set to '1'). I created myseld with command line and ensure it was then created in data directory.
Did i miss something?
